src/VisualStudio/Core/Def/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -149,13 +152,27 @@ internal sealed class SourceGeneratedFileManager : IOpenTextBufferEventListener
 
             identity = default;
 
+            // Find GUID in "temporary directory/GUID/path/to/file.cs".
             if (!filePath.StartsWith(_temporaryDirectory))
```

Comment:
don't love all these case sensitive types of checks.  I feel like this should all be using normal Path operations, or Roslyn's PAth impls (that do the right thing on varying file systems).

Reply:
These never exist on disk though, right? Even when we do write them to disk, it's part of a later step and not part of the main algorithm.

Reply:
VS for Windows doesn't have support for virtual files, so this file here is empty file we do write when a file is actually opened.

Reply:
Would love to use `Path` or `PathUtilities`, but haven't found any way to extract the GUID from path like "temporary directory/GUID/path/to/file.cs" except maybe by looping over `new DirectoryInfo().Parent` which seems excessive.